crates/compiler/gen_dev/src/generic64/x86_64.rs

```diff
@@ -1465,6 +1548,12 @@ fn imul_reg64_reg64(buf: &mut Vec<'_, u8>, dst: X86_64GeneralReg, src: X86_64Gen
     extended_binop_reg64_reg64(0x0F, 0xAF, buf, src, dst);
 }
 
+/// `MUL r/m64` -> Unsigned Multiply r/m64 to r64.
+#[inline(always)]
+fn mul_reg64_reg64(buf: &mut Vec<'_, u8>, dst: X86_64GeneralReg, src: X86_64GeneralReg) {
+    todo!()
+}
```

Comment:
this is where I got stuck with `mul` being different from `imul` because `mul` takes just one argument where `imul` takes two

Reply:
From the opcode `REX.W + F7 /4`
I would guess this should work:
`binop_reg64_reg64(0xF7, buf, X86_64GeneralReg::RAX, src);`

Reply:
we're not really dealing with a binop though. Is that helper still correct?